### PR TITLE
PMI verification tests

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17808)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/269071613)

File: payment-request/updateWith-method-pmi-handling-manual.https.html [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  currency: "USD",
+  value: "1.0",
+});
+
+const validTotal = Object.freeze({
+  label: "Default Total",
+  amount: validAmount,
+});
+
+const validShippingOption = {
```
Comment (marcoscaceres) [inline payment-request/updateWith-method-pmi-handling-manual.https.html]: should freeze

File: payment-request/updateWith-method-pmi-handling-manual.https.html [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  amount: validAmount,
+  selected: true,
+};
+
+const validDetails = Object.freeze({
+  total: validTotal,
+  shippingOptions: [validShippingOption],
+});
+
+const validModifier = [
```
Comment (marcoscaceres) [inline payment-request/updateWith-method-pmi-handling-manual.https.html]: take out of array

File: payment-request/updateWith-method-pmi-handling-manual.https.html [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+// Avoid false positives, this should always pass
+function smokeTest() {
+  new PaymentRequest(validMethods, validDetails);
+}
+
+function manualTest(button, { invalidMethod }) {
+  button.disabled = true;
+  promise_test(async t => {
+    smokeTest();
```
Comment (marcoscaceres) [inline payment-request/updateWith-method-pmi-handling-manual.https.html]: should just inline this

File: payment-request/payment-request-ctor-pmi-handling.https.html [hunk 4]
```diff
@@ -32,6 +32,15 @@
...
   const invalidMethods = [
     {
+      supportedMethods: "https://username@example.com/pay",
+    },
+    {
+      supportedMethods: "https://:password@example.com/pay",
+    },
+    {
+      supportedMethods: "https://username:password@example.com/pay",
+    },
```
Comment (marcoscaceres) [inline payment-request/payment-request-ctor-pmi-handling.https.html]: add a test with query string

**Review by edenchuang — APPROVED** [review]
LGTM

**Review by edenchuang — CHANGES_REQUESTED** [review]
Sorry, didn't see the newest commit.

**Comment by marcoscaceres** [comment]
@edenchuang no problem at all. Will fix.

File: payment-request/updateWith-method-pmi-handling-manual.https.html [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            supportedMethods: invalidMethod,
+          });
+          const invalidDetails = Object.assign({}, validDetails, {
+            modifiers: [validModifier, invalidModifier],
+          });
+          try {
+            ev.updateWith(invalidDetails);
+            resolve(); // This is bad.
+          } catch (err) {
+            reject(err); // this is good.
```
Comment (edenchuang) [inline payment-request/updateWith-method-pmi-handling-manual.https.html]: I am not sure if this is correct or not. Promise reject/resolve callback might not be executed synchronized, if it is not, updateWith() might not emit any errors and finish its work before the callbacks were called.

**Review by edenchuang — CHANGES_REQUESTED** [review]

File: payment-request/updateWith-method-pmi-handling-manual.https.html [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+          }
+        },
+        { once: true }
+      );
+    });
+    const showPromise = request.show();
+    await promise_rejects(t, new RangeError(), shippingChangedPromise);
+    await promise_rejects(t, "AbortError", showPromise);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply payment-request/updateWith-method-pmi-handling-manual.https.html]: Correct. Fixing.

File: payment-request/updateWith-method-pmi-handling-manual.https.html (hunk 5 above)
Comment (marcoscaceres) [reply payment-request/updateWith-method-pmi-handling-manual.https.html]: yeah, you are correct here also.

**Review by edenchuang — APPROVED** [review]
LGTM

**Pull request merged**